api/types/src/error.rs

```diff
@@ -42,15 +53,24 @@ impl Error {
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}: {}", self.status_code(), &self.message)
+        match &self.data {
+            Some(val) => write!(f, "{}: {}\n{}", self.status_code(), &self.message, &val),
+            None => write!(f, "{}: {}", self.status_code(), &self.message),
+        }
```

Comment:
If you want to avoid the duplication you can do:

```
write!(f, "{}: {}", self.status_code(), &self.message)?;
if let Some(val) = &self.data {
write!(f, "\n{}", val)?;
}
Ok(())
```